docs/InstrumentationEvents.md

```diff
@@ -25,74 +25,29 @@ Instrumentation Event:
+|4      | FETCH_START         | {}                                                                                                                                                                                                            | Starting to fetch messages from broker |
+|5      | FETCH               | {`numberOfBatches`, `duration`}                                                                                                                                                                           | Fetched messages from the broker |
+|6      | START_BATCH_PROCESS | {`topic`, `partition`, `highWatermark`, `offsetLag`, `offsetLagLow`, `batchSize`, `firstOffset`, `lastOffset`}                                                                                | Starting to process the batch |
+|7      | END_BATCH_PROCESS   | {`topic`, `partition`, `highWatermark`, `offsetLag`, `offsetLagLow`, `batchSize`, `firstOffset`, `lastOffset`, `duration`}                                                                  | Processed the batch |
+|8      | COMMIT_OFFSETS      | {`groupId`, `memberId`, `groupGenerationId`, `topics`}                                                                                                                                                | Committed resolved offsets |
+|9     | STOP                |                                                                                                                                                                                                               | Consumer has stopped |
+|10     | DISCONNECT          |                                                                                                                                                                                                               | Consumer has disconnected |
+|-      | CRASH               | {`error`, `groupId`}                                                                                                                                                                                      | Consumer has crashed. After this, consumer will try to perform a full restart if possible, otherwise it will stay hanging. Ideally you should listen to this event and handle yourself. |
+|-      | HEARTBEAT           | {`groupId`, `memberId`, `groupGenerationId`}                                                                                                                                                            | Heartbeat sent to the broker |
+|-      | REQUEST_TIMEOUT     | {`broker`, `clientId`, `correlationId`, `createdAt`, `sentAt`, `pendingDuration`, `apiName`, `apiKey`, `apiVersion`}                                 | Request to the broker timed out |
+|-      | REQUEST_QUEUE_SIZE  | {`broker`, `clientId`, `queueSize`}                                                                                                                                                      | ?? Too many requests? |
```

Comment:
All requests go through a request queue where concurrency is managed (`maxInflightRequests`). Whenever the size of the queue changes, this event is emitted.